0011-command-execution-protocol/README.md

```diff
@@ -0,0 +1,170 @@
+    }
+  },
+  "additionalProperties": false
+}
+```
+
+If `observe` field is present and is `true`, the server MUST send new commands to the endpoint whenever they arrive.
+
+If `observe` field is present and is `false`, the server MUST NOT send new commands to the endpoint whenever they arrive.
+
+The server MAY send commands to the endpoint when no request were done before, or `observe` was not present in a request.
+
+The server MAY support subscribing to all command types by accepting requests at `/<endpoint_token>/command` resource path. In that case, command representation MUST include `type` property that is a string identifying the command type.
+
+### Response
+The server MUST respond with all outstanding commands of the specified command type.
+
+The response MUST be a UTF-8 encoded array with the schema defined in [command-response.schema.json](./command-response.schema.json) file.
+
+```json
+{
+  "$schema": "http://json-schema.org/schema#",
+  "title": "11/CEP command response schema",
+
+  "type": "array",
+  "items": {
+    "type": "object",
+    "properties": {
+      "id": {
+        "type": "integer",
+        "description": "ID of the command."
+      },
+      "payload": {
+        "description": "The command payload to be interpreted by the endpoint."
+      }
+    },
+    "required": [
+      "id"
+    ],
+    "additionalProperties": false
+  }
+}
+```
+
+Each element of the array represents an outstanding command that hasn't been completed or expired yet.
+
+`id` property is used to match command results to commands and MUST be unique within the command type and an endpoint.
+
+When there is no outstanding commands, the server MUST return an empty array.
+
+Upon reception of a command, the endpoint SHOULD "execute" the command in a way that is command-specific and is out of scope of this document. Upon a command execution success or failure, the endpoint MUST push the command result using the Result resource.
+
+## Result resource
+The result resource is used to notify the server of the command execution status (success/failure) and results.
+
+Result resource is a request/response resource with the following resource path:
+```
+/<endpoint_token>/result/<command_type>
+```
+where `<command_type>` is a type of the command.
+
+### Request
+The request is a UTF-8 encoded JSON object with the JSON Schema defined in the [result-request.schema.json](./result-request.schema.json) file.
+
+```json
+{
+  "$schema": "http://json-schema.org/schema#",
+  "title": "11/CEP result request schema",
+
+  "type": "object",
+  "properties": {
+    "id": {
+      "type": "integer",
+      "description": "ID of the command."
+    },
+    "statusCode": {
+      "type": "integer",
+      "description": "Status code of the command execution. Based on HTTP status codes."
+    },
+    "reasonPhrase": {
+      "type": "string",
+      "description": "A human-readable string explaining the cause of an error if any."
```

Comment:
It kind of is. It is optional as I don't see a reason to send `reasonPhrase` on success or when there is nothing interesting to say.